java/src/jmri/web/servlet/frameimage/JmriJFrameServlet.java

```diff
@@ -207,6 +207,54 @@ private void sendClickSequence(MouseListener m, Component c, int x, int y) {
 
     @Override
     protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        // because we work with Swing, we do this on the AWT thread
+        
+        if (javax.swing.SwingUtilities.isEventDispatchThread()) {
+            doGetOnSwing(request, response);
+            return;
+        }
+        
+        try {
+            javax.swing.SwingUtilities.invokeAndWait(
+                () -> { 
+                    try {
+                        doGetOnSwing(request, response);
+                    } catch ( ServletException | IOException ex ) {
+                        throw new RuntimeException(ex);
```

Comment:
Why do you catch these exceptions and throw a RuntimeException instead?

Reply:
You can't throw ServletException or IOException out of Runnable#run() because it's not been declared there.